### ISSUE TITLE
[Firefox Mobile] blocked requests count keeps constantly increasing when navigating github

### ISSUE THREAD

**Issue by WalterWW** [uBlock-LLC/uBlock#1071]
Steps to reproduce:
1. Open github issues page. Number of blocked requests = 4. http://i.imgur.com/ksrkD3q.jpg
2. Open a random issue from the list, then hit the back button. Do this 10 times. Number of blocked requests now = 33. http://i.imgur.com/He7b511.jpg

Maybe this is actually intented behavior though I don't know.

**Comment by gorhill** [comment]
Duplicate of #722. Github builds its pages incrementally most of the time, i.e. not by pulling in a whole new document. The counter is reset only when a full page reload occurs.

**Comment by WalterWW** [comment]
Okay, that explains it. Would maybe resetting the blocking stats when using the back button make sense to mitigate this?

**Comment by gorhill** [comment]
> Would maybe resetting the blocking stats when using the back button make sense to mitigate this?

No. When we go backward, there is no document fetched. From our point of view, it does appear that a new document is fetched and that stats should be reset, but really this is all incremental. Imagine the case where only a very small area of the page gets updated: in such case, it would not make sense that the stats reset whenever that small area gets updated. Github just happens to update such a large area that it almost appears as it is a page reload.

Bottom line is that uBlock sees incremental update of a page, and when this happens the semantic is that it's not a full page reload, and stats have to stick. Doing otherwise is not welcomed, or else you would end up with cases like stats on Google Maps resetting every time you pan or zoom.

**Comment by WalterWW** [comment]
Thanks for the explanation.
